ingest/cell_metadata.py

```diff
@@ -74,38 +74,48 @@ def create_subdocuments(self):
         for value in self.headers[1:]:
             # Copy subdocument model so memory references are different
             copy_of_subdoc_model = copy.copy(
-                {"cell_names": self.cell_names, "values": []}
+                {'cell_names': self.cell_names, 'values': []}
             )
             sub_documents[value] = copy_of_subdoc_model
         return sub_documents
 
     def get_collection_name(self):
         """Returns collection name"""
-        return "cell_metadata"
+        return 'cell_metadata'
 
     def get_subcollection_name(self):
         """Returns sub-collection name"""
-        return "data"
+        return 'data'
+
+    def store_validation_error(self, type, category, msg, associated_info=None):
+        """Store validation errors in proper arrangement
+        :param type: type of error (error or warn)
```

Comment:
@jlchang I used warnings what I was trying to convert cell metadata to pandas. Take a look in my [branch](https://github.com/broadinstitute/scp-ingest-pipeline/blob/0b8f8c014544dd1c9e8fa9e9b419fb764338e83d/ingest/cell_metadata.py#L82). This would address @eweitz concerns

Reply:
@knapii-developments I like the use of warnings but because we eventually want to provide file-based output of errors and warnings, I need to collect both warnings and errors for later use. I've updated the code to refer to errors and warnings collectively as "issues" per Eric's suggestion by changing CellMetadata to have an "issues" attribute instead of a "errors" attribute and propagated the "errors -> issues" change throughout as appropriate.

Reply:
@jlchang I think you should check out logging for [warnings module](https://docs.python.org/3/library/logging.html#integration-with-the-warnings-module) that python offers. I think it may be able to fulfill your needs. Here's some [examples](https://programtalk.com/python-examples-amp/logging.captureWarnings/) that may be useful.